### PR TITLE
feat: implement EncryptionService unit tests and fix test encryption key

### PR DESCRIPTION
**Pull request by saksham869**
## Summary
Added unit tests for EncryptionService and fixed a broken test encryption key configuration.

### 1. Fixed test encryption key (src/test/resources/application.properties)
- The old key (1234567890123456) was a raw string, but EncryptionService calls Base64.getDecoder().decode(key), so a proper Base64-encoded key is required
- Replaced with a valid Base64-encoded 32-byte AES-256 key

### 2. Implemented 7 unit tests (EncryptionServiceUnitTest.java)
Addresses the //TODO Implement in the file:

| Test | What it verifies |
|------|-----------------|
| encryptReturnsNonNullBase64String | Basic encrypt works, output is valid Base64 |
| decryptReturnsOriginalPlaintext | Encrypt to Decrypt round-trip correctness |
| encryptProducesDifferentCiphertextEachTime | Random IV ensures unique ciphertexts |
| decryptWithTamperedCiphertextThrowsException | GCM authentication tag detects tampering |
| encryptAndDecryptWithSpecialCharacters | Unicode, symbols, multi-byte characters |
| encryptAndDecryptEmptyString | Edge case: empty string |
| encryptAndDecryptLongString | Stress test with long API key strings |

## Test Results
All tests pass:
- EncryptionServiceUnitTest: 7/7 passed
- CreditBureauRegistrationServiceIntegrationTest: 4/4 passed (6 pre-existing @Disabled stubs unchanged)

## Summary by CodeRabbit

* **Tests**
  * Expanded encryption service coverage with a full JUnit 5 suite verifying Base64 output, successful decrypt round-trips, randomized ciphertext (IV), tamper detection, and correctness for empty, Unicode, and long inputs.

* **Chores**
  * Updated the test encryption configuration to use a fixed hardcoded key in the test properties file.

### CODE REVIEW

**Review by yuwatidora — CHANGES_REQUESTED** [review]

File: src/test/java/org/mifos/creditbureau/service/EncryptionServiceUnitTest.java [hunk 1]
```diff
@@ -1,5 +1,127 @@
...
+ * <p>
+ * Verifies AES-256-GCM encryption and decryption behaviour including:
+ * round-trip correctness, IV randomness, tamper detection, and edge cases.
+ */
+class EncryptionServiceUnitTest {
+
+    private EncryptionService encryptionService;
+
+    // Base64-encoded 32-byte AES-256 key (matches test application.properties)
+    private static final String TEST_KEY = "QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVoxMjM0NTY=";
```
[the remark this answers is not in the document]
Comment (saksham869) [reply src/test/java/org/mifos/creditbureau/service/EncryptionServiceUnitTest.java:21]: Hi @yuwatidora, done! I've updated both files as suggested. However, CodeRabbit flagged that ${MIFOS_SECURITY_ENCRYPTION_KEY} in test application.properties will fail if the env var isn't set at test time. Should I use @TestPropertySource to inject a fixed test key, or keep it as is and set the env var when running tests? Happy to follow your guidance

**Comment by saksham869** [comment]
Hi @yuwatidora ,
I have addressed all the review feedback:

1. Removed @SpringBootTest — EncryptionService now instantiated directly in @BeforeEach with a fixed TEST_KEY. No Spring context needed.
2. Fixed weak assertion in encryptProducesDifferentCiphertextEachTime — both ciphertexts now asserted to equal the original plaintext, not just each other.
3. Replaced ${MIFOS_SECURITY_ENCRYPTION_KEY} in test application.properties with a hardcoded test key so tests run without any environment variable set.

Please let me know if anything else needs to be changed.

Thank you!
